packages/cli/src/commands/hydrogen/upgrade.ts

```diff
@@ -0,0 +1,2931 @@
+  const pinnedCurrentVersion = getAbsoluteVersion(currentVersion);
+  const currentReleaseIndex = changelog.releases.findIndex((release) => {
+    const pinnedReleaseVersion = getAbsoluteVersion(release.version);
+    return pinnedReleaseVersion === pinnedCurrentVersion;
+  });
+
+  const uniqueNextReleases = changelog.releases
+    .slice(0, currentReleaseIndex)
+    .reverse()
+    .reduce((acc, release) => {
+      if (acc[release.version]) return acc;
+      acc[release.version] = release;
+      return acc;
+    }, {} as Record<string, Release>);
+
+  const nextReleases = Object.keys(uniqueNextReleases).length
+    ? Object.entries(uniqueNextReleases)
+        .map(([version, release]) => {
+          return `${version} - ${release.title}`;
+        })
+        .slice(0, 5)
+    : [];
+
+  let headline =
+    Object.keys(uniqueAvailableUpgrades).length > 1
+      ? `There are ${
+          Object.keys(uniqueAvailableUpgrades).length
+        } new @shopify/hydrogen versions available.`
+      : `There's a new @shopify/hydrogen version available.`;
+
+  renderInfo({
+    headline,
+    body: [`Current: ${currentVersion} | Latest: ${pinnedLatestVersion}`],
+    //@ts-ignore will always be an array
+    customSections: nextReleases.length
+      ? [
+          {
+            title: `The next ${nextReleases.length} version(s) include`,
+            body: [
+              {
+                list: {
+                  items: [
+                    ...nextReleases,
+                    availableUpgrades.length > 5 && `...more`,
+                  ]
+                    .flat()
+                    .filter(Boolean),
+                },
+              },
+            ].filter(Boolean),
+          },
+          {
+            title: 'Next steps',
+            body: [
+              {
+                list: {
+                  items: [
+                    `Run \`h2 upgrade\` or \`h2 upgrade --version XXXX.X.XX\``,
+                    ,
+                    `Read release notes at https://hydrogen.shopify.dev/releases`,
+                  ],
+                },
+              },
+            ],
+          },
+        ]
+      : [],
+  });
+}
+
+/**
+ * Temporary snapshot of the changelog.json file
+ * Once we merge to production, we can remove this and use the fetchChangelog function
+ */
+const TEMP_CHANGELOG: ChangeLog = {
+  url: 'https://github.com/Shopify/hydrogen/pulls?q=is%3Apr+is%3Aclosed+%5Bci%5D+release+in%3Atitle+is%3Amerged',
+  version: '1',
+  releases: [
+    {
+      title: 'Moved @remix-run/server-runtime to peer dependency',
+      version: '2023.10.2',
+      date: '',
+      hash: '875d35f52fcfd8006e78217cfac0096122f98a54',
+      pr: 'https://github.com/Shopify/hydrogen/pull/1486',
+      commit:
+        'https://github.com/Shopify/hydrogen/pull/1486/commits/875d35f52fcfd8006e78217cfac0096122f98a54',
+      dependencies: {
+        '@remix-run/react': '2.1.0',
```

Comment:
Are we going to support optional dependencies from now or add it later?
If we do it later, then maybe we should add them to a separate field like `dependenciesMeta` to avoid breaking changes. Otherwise, if we want to at least make it compatible  now, we should start reading versions like `dependencies[x]?.version || dependencies[x]`, or something like that :thinking:

Reply:
[Follow](https://docs.npmjs.com/cli/v10/configuring-npm/package-json#peerdependenciesmeta)